iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java

```diff
@@ -20,43 +20,62 @@
 package org.sonar.iac.kubernetes.checks;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
+
 import org.sonar.check.Rule;
 import org.sonar.iac.common.api.checks.CheckContext;
 import org.sonar.iac.common.api.checks.SecondaryLocation;
 import org.sonar.iac.common.yaml.object.BlockObject;
 import org.sonar.iac.common.yaml.tree.MappingTree;
 import org.sonar.iac.common.yaml.tree.ScalarTree;
+import org.sonar.iac.common.yaml.tree.TupleTree;
 import org.sonar.iac.common.yaml.tree.YamlTree;
+import org.sonar.iac.kubernetes.model.ConfigMap;
+import org.sonar.iac.kubernetes.model.MapResource;
+import org.sonar.iac.kubernetes.model.Secret;
 import org.sonar.iac.kubernetes.visitors.KubernetesCheckContext;
 
 @Rule(key = "S6907")
-public class DuplicatedEnvironmentVariablesCheck extends AbstractKubernetesObjectCheck {
+public class DuplicatedEnvironmentVariablesCheck extends AbstractResourceManagementCheck<MapResource> {
 
-  private static final String MESSAGE = "Resolve the duplication of this environment variable.";
-  private static final String MESSAGE_SECONDARY_LOCATION = "Duplicate environment variable without any effect.";
+  private static final Map<Class<? extends Variable>, String> PRIMARY_MESSAGE_PER_KIND = Map.of(
+    EnvVariable.class, "Resolve the duplication of this environment variable.",
+    ConfigMapVariable.class, "Resolve the duplication of the environment variable '%s' in this ConfigMap.",
+    SecretVariable.class, "Resolve the duplication of the environment variable '%s' in this Secret.");
+  private static final Map<Class<? extends Variable>, String> SECONDARY_MESSAGE_PER_KIND = Map.of(
+    EnvVariable.class, "Duplicate environment variable without any effect.",
+    ConfigMapVariable.class, "ConfigMap that contain the duplicate environment variable '%s' without any effect.",
+    SecretVariable.class, "Secret that contain the duplicate environment variable '%s' without any effect.");
```

Comment:
Instead of such maps maybe it will be better to add methods `primaryMessage(String variableName)` / `secondaryMessage(String variableName)` to all implementations of `Variable`?
Then, we will benefit from the inheritance mechanism instead of using such a strange `Map`s.